PennMobile/General/Protocols.swift

```diff
@@ -145,3 +146,41 @@ extension LocationPermissionRequestable {
         }
     }
 }
+
+protocol LocalAuthentication {
+    func handleAuthenticationSuccess()
+    func handleAuthenticationFailure()
+}
+
+extension LocalAuthentication {
+    
+    func handleAuthentication(cancelText : String, reasonText: String) {
+        let context = LAContext()
+        context.localizedCancelTitle = cancelText
+
+        // Check if we have the needed hardware support.
+        var error: NSError?
+        if context.canEvaluatePolicy(.deviceOwnerAuthentication, error: &error) {
```

Comment:
Under what circumstance would the user not have the needed hardware support?

Reply:
It is possible for users to disable both Touch ID and FaceID on their device. The `.canEvaluatePolicy` checks if the we can access biometrics before calling the  `canEvaluatePolicy(_:error:)` that might crash the app without the check. It is mentioned in the **Discussion** portion of this [documentation](https://developer.apple.com/documentation/localauthentication/lacontext/1514149-canevaluatepolicy).